app/api/sections.js

```diff
@@ -56,4 +93,120 @@ router.post('/:course_id/sections', requireAuthentication, async function (req,
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const courseId = parseInt(req.params['course_id'])
+
+    const isTeacher = await checkIfTeacher(user.id, courseId)
+
+    if (isTeacher) {
+        try {
+            const sectionsInCourse = await getSectionsInCourse(courseId)
+            if (sectionsInCourse.length != 0) {     // if sections are present for this course
+                res.status(200).send(sectionsInCourse)
+            }
+            else {
+                res.status(204).send()  // no sections for this course
+            }
+        }
+        catch (e) {
+            if (e instanceof ValidationError) {
+                return res.status(400).send({error: serializeSequelizeErrors(e)})
+            }
+            else {
+                next(e)
+            }
+        }
+    }
+    else {
+        res.status(403).send({error: "Only the teacher for a course can get a section"})
+    }
+})
+
+// get a specific section, and lectures for that section
+router.get('/:course_id/sections/:section_id', requireAuthentication, async function (req, res, next) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const courseId = parseInt(req.params['course_id'])
+    const sectionId = parseInt(req.params['section_id'])
+
+    const isTeacher = await checkIfTeacher(user.id, courseId)
+    const respObj = {}  // will hold the full endpoint response at the end
+
+    if (isTeacher) {
+        try {
+            const foundSection = await getSection(sectionId)
+            if (foundSection != null) {     // if section was found for given id
+                respObj['section'] = foundSection
+                const relatedLectures = await getLecturesFromSection(sectionId)   // get lectures and append to response
+                respObj['lectures'] = relatedLectures
+                res.status(200).send(respObj)
+            }
+            else {
+                res.status(404).send({error: "There is no section for the provided section id"})
+            }
+        }
+        catch (e) {
+            if (e instanceof ValidationError) {
+                return res.status(400).send({error: serializeSequelizeErrors(e)})
+            }
+            else {
+                next(e)
+            }
+        }
+    }
+    else {
+        res.status(403).send({error: "Must be a teacher of this course to get section info"})
+    }
+})
+
+// update a specific section
+router.put('/:course_id/sections/:section_id', requireAuthentication, async function (req, res, next) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const courseId = parseInt(req.params['course_id'])
+    const sectionId = parseInt(req.params['section_id'])
+    const newNumber = req.body.number
+
+    const isTeacher = await checkIfTeacher(user.id, courseId)
+    const foundSection = await getSection(sectionId)
```

Comment:
I think we need to make sure this section belongs to the course the teacher is in